src/components/VDataTable/mixins/body.js

```diff
@@ -57,13 +57,14 @@ export default {
       return rows
     },
     genEmptyItems (content) {
-      if (typeof content === 'string') {
+      if (typeof content === 'string' || (!this.hasTR(content) && !this.needsTR(content))) {
```

Comment:
Generating TR when `!needsTR()` was a bit confusing for me when reading the code. I'd replace `hasTR` and `needsTR` with `hasTag(tag1, tag2, ...)`:
```js
if (hasTag('tr')) {
  return content
} else if (hasTag('td', 'th')) {
   return genTR(content)
} else {
  return genTR(genTD(...))
}
```